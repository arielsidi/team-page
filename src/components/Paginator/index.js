//@flow
import React from "react";
import styled from "styled-components";

import { createPages } from "../../util/pagination";

const Pages = styled.div`
    display: inline-block;
    border: 1px solid ${props => props.theme.appBgColor};

    a {
        color: ${props => props.theme.primaryColor};
        font-size: 15px;
        float: left;
        padding: 8px 13px;
        text-decoration: none;
        transition: background-color 0.3s;
    }

    a:not(:first-child) {
        border-left: 1px solid ${props => props.theme.appBgColor};
    }

    a.active {
        background-color: ${props => props.theme.primaryColor};
        color: white;
        border: 1px solid ${props => props.theme.primaryColor};
    }

    a.disabled {
        color: ${props => props.theme.textColor};
        opacity: 0.7;
    }

    a:hover:not(.active):not(.disabled) {
        background-color: ${props => props.theme.appBgColor};
    }

    a.active,
    a.disabled {
        cursor: default;
    }
`;

type PropsType = {
    pageCount: number,
    currentPage: number,
    pageChange: (page: number) => void
};

const Paginator = ({ pageCount, currentPage, pageChange }: PropsType) => {
    pageCount = Number(pageCount);
    currentPage = Number(currentPage);
    const pages = createPages(pageCount, currentPage);

    return (
        <Pages>
            <a
                href="#"
                id="prevPageButton"
                {...(currentPage == 1 && { className: "disabled" }) || {
                    onClick: () => pageChange(currentPage - 1)
                }}
            >
                &lt;
            </a>
            {pages.map((number, index) => (
                <a
                    href="#"
                    key={number}
                    {...(currentPage == number && { className: "active" }) || {
                        onClick: () => pageChange(number)
                    }}
                >
                    {(number !== 1 && index == 0 && "..") ||
                        (number !== pageCount &&
                            index == pages.length - 1 &&
                            "..") ||
                        number}
                </a>
            ))}
            <a
                href="#"
                id="nextPageButton"
                {...(currentPage == pageCount && { className: "disabled" }) || {
                    onClick: () => pageChange(currentPage + 1)
                }}
            >
                &gt;
            </a>
        </Pages>
    );
};

export default Paginator;
