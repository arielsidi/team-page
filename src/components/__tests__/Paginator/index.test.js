import React from "react";
import Paginator from "../../Paginator";
import renderer from "react-test-renderer";

describe("Paginator", () => {
    it("Should render a paginator with 10 pages and current page 2.", () => {
        const tree = renderer
            .create(<Paginator pageCount="10" currentPage="2" />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
