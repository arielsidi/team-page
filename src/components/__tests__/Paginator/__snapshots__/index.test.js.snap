// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator Should render a paginator with 10 pages and current page 2. 1`] = `
<div
  className="Paginator__Pages-sc-1yt4i60-0 kHWZPw"
>
  <a
    href="#"
    onClick={[Function]}
  >
    &lt;
  </a>
  <a
    href="#"
    onClick={[Function]}
  >
    1
  </a>
  <a
    className="active"
    href="#"
  >
    2
  </a>
  <a
    href="#"
    onClick={[Function]}
  >
    3
  </a>
  <a
    href="#"
    onClick={[Function]}
  >
    4
  </a>
  <a
    href="#"
    onClick={[Function]}
  >
    5
  </a>
  <a
    href="#"
    onClick={[Function]}
  >
    6
  </a>
  <a
    href="#"
    onClick={[Function]}
  >
    ..
  </a>
  <a
    href="#"
    onClick={[Function]}
  >
    &gt;
  </a>
</div>
`;
